workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/DockerValidator.java

```diff
@@ -38,9 +38,17 @@
 
 	private boolean isDockerAvailable = false;;
 
+	/**
+	 * 1. It will check that spec has buildspec or dockerfile 
+	 * 2. If both is not then it will return true
+	 * 3. If any one is there then 
+	 *    3.1  It will verify that docker is installed or not
+	 *    3.2  It will run docker command 
+	 *    3.3  if command executed successfully then return true else false
+	 */
 	@Override
 	public boolean validate(WorkflowContext context) throws HyscaleException {
-		if (!ImageDetailsUtil.isImageBuildPushRequired(context.getServiceSpec())) {
+		if (!ImageUtil.isImageBuildPushRequired(context.getServiceSpec())) {
 			return true;
 		}
 		if (isDockerAvailable) {
```

Comment:
This should be the first check, before build push required check

Reply:
If the validation is already checked , just return the result.